test/test_helper.rb

```diff
@@ -110,6 +113,58 @@ def fake_machine(options = {})
   machine
 end
 
+class FakeHTTPProxyServer < WEBrick::HTTPProxyServer
+  def initialize(url, config)
+    @url = url
+    super(config)
+  end
+
+  def service(req, res)
+    req.attributes['url'] = @url
+    super(req, res)
+  end
+end
+
+module WEBrick
```

Comment:
> Just provide a logger.

Tried that solution but it seems to be not working. Weird behavior is it is only happening with authenticated proxy server.
See the other logs are going to StringIO instance.